api/src/main/java/org/apache/iceberg/DeleteMarkersScanTask.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+/**
+ * A scan task for delete markers generated by adding equality delete files to the table.
+ * <p>
+ * Note that such tasks are created only if equality deletes are not assigned to data files
+ * and are used purely for producing delete markers.
+ */
+public interface DeleteMarkersScanTask extends ChangelogScanTask {
```

Comment:
This task is a little bit weird as it would wrap an equality delete file as `DataFile`. We do that somewhere else but it may be a good precedent to reconsider extending `FileScanTask`.

Reply:
why is this interface needed in addition to `DeletedRowsScanTask`?

Reply:
I wanted to handle cases when we should output the content of equality deletes as-is. That said, I'll probably skip this for now to simplify the initial implementation and we can reconsider it back.

Reply:
I removed this task type for now.